lib/utils/print_pdf.dart

```diff
@@ -143,19 +143,42 @@ class PrintService {
   }
 
   _addTotalLineToTable(Map<String, ByteData> fontMap) {
-    double totalAmount = printParameters.submittedSale.totalChargeSat /
-        printParameters.currentCurrency.satConversionRate;
+    final sale = printParameters.submittedSale;
```

Comment:
Small one:
The PrintService class supposed to do as less calculation and business logic as possible and concentrate at printing.
It seems that moving out the logic of calculating the totals outside this class would be compatible with this approach. This can be done by adding these methods on the Sale class and use them from here:
1.  A method to calculate totalAmount by by currency
2. A method to get the single currency if exists

Reply:
makes all sense, I'll change it, thank you @roeierez